synapse/storage/databases/main/event_federation.py

```diff
@@ -970,9 +1001,40 @@ def _get_backfill_events(self, txn, room_id, event_list, limit):
 
             event_results.add(event_id)
 
+            txn.execute(
+                connected_insertion_event_query, (event_id, limit - len(event_results))
+            )
+            connected_insertion_event_id_results = txn.fetchall()
+            logger.debug(
+                "_get_backfill_events: connected_insertion_event_query %s",
+                connected_insertion_event_id_results,
+            )
```

Comment:
Can you add a comment about what this is all doing at a high level? i.e. something like "if we have an insertion event pointing at this event then we need to go and pull in all connected chunks", but with better words